Server/mods/deathmatch/logic/CPlayerManager.cpp

```diff
@@ -74,327 +67,139 @@ void CPlayerManager::PulseZombieCheck()
 CPlayer* CPlayerManager::Create(const NetServerPlayerID& PlayerSocket)
 {
     // Check socket is free
-    CPlayer* pOtherPlayer = MapFindRef(m_SocketPlayerMap, PlayerSocket);
-    if (pOtherPlayer)
+    if (CPlayer* pOtherPlayer = Get(PlayerSocket))
     {
         CLogger::ErrorPrintf("Attempt to re-use existing connection for player '%s'\n", pOtherPlayer->GetName().c_str());
         return NULL;
     }
 
     // Create the new player
-    return new CPlayer(this, m_pScriptDebugging, PlayerSocket);
-}
-
-unsigned int CPlayerManager::CountJoined()
-{
-    // Count each ingame player
-    unsigned int                   uiCount = 0;
-    list<CPlayer*>::const_iterator iter = m_Players.begin();
-    for (; iter != m_Players.end(); iter++)
-    {
-        if ((*iter)->IsJoined())
-        {
-            ++uiCount;
-        }
-    }
-
-    // Return the count
-    return uiCount;
-}
-
-bool CPlayerManager::Exists(CPlayer* pPlayer)
-{
-    return m_Players.Contains(pPlayer);
-}
-
-CPlayer* CPlayerManager::Get(const NetServerPlayerID& PlayerSocket)
-{
-    return MapFindRef(m_SocketPlayerMap, PlayerSocket);
-}
-
-CPlayer* CPlayerManager::Get(const char* szNick, bool bCaseSensitive)
-{
-    // Try to find it in our list
-    const char*                    szTemp;
-    list<CPlayer*>::const_iterator iter = m_Players.begin();
-    for (; iter != m_Players.end(); iter++)
-    {
-        // Grab the nick pointer
-        szTemp = (*iter)->GetNick();
-        if (szTemp)
-        {
-            // Do they equal?
-            if ((bCaseSensitive && strcmp(szNick, szTemp) == 0) || (!bCaseSensitive && stricmp(szNick, szTemp) == 0))
-            {
-                return *iter;
-            }
-        }
-    }
-
-    // If not, return NULL
-    return NULL;
+    return new CPlayer(this, m_pScriptDebugging, PlayerSocket); // Seems very safe
```

Comment:
```
// Seems very safe
```
Please don't add such comments, what is this supposed to mean?

Reply:
I should've added `TODO: Seems very safe`.
It means that this should be cleaned up, so that the manager manages the lifetime of the CPlayer instances.